### PR TITLE
feat: add example for OpenAPI and mocking

### PR DESCRIPTION
**Pull request by fiam**

### CODE REVIEW

**Comment by StarpTech** [comment]
Please add the example to the index on the readme.

**Comment by fiam** [comment]
> Please add the example to the index on the readme.

I had totally missed that list. Adding it, thanks!

File: examples/openapi-mocking/.wundergraph/api.yaml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+servers:
+  - url: http://api.example.com/v1
+    description: Example Server
+
+paths:
+  /notes/note/{noteID}:
+    get:
+      summary: Returns a note by ID
+      parameters:
```
Comment (jensneuse) [inline examples/openapi-mocking/.wundergraph/api.yaml:15]: add the property "operationId" (type string) to give this field a better GraphQL-like name, e.g. "noteByID"

Also add a comment to the README and docs website about this.
It'll help users get a better generated GraphQL Schema.

File: examples/openapi-mocking/.wundergraph/api.yaml [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        '400':
+          description: The specified note ID is invalid (not a number).
+        '404':
+          description: A note with the specified ID was not found.
+        default:
+          description: Unexpected error
+  /notes/new:
+    post:
+      summary: Creates a new note
+      parameters:
```
Comment (jensneuse) [inline examples/openapi-mocking/.wundergraph/api.yaml:46]: Add "operationId" here as well.

File: examples/openapi-mocking/.wundergraph/api.yaml (hunk 1 above)
Comment (jensneuse) [reply examples/openapi-mocking/.wundergraph/api.yaml:15]: Example: https://github.com/wundergraph/wundergraph/blob/main/testapps/default/json_placeholder.json#L18

Comment (jensneuse) [reply examples/openapi-mocking/.wundergraph/api.yaml:15]: Don't forget to re-generate the code and update the Operations after this change.

File: examples/openapi-mocking/.wundergraph/wundergraph.server.ts [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+	hooks: {
+		queries: {
+			NoteByID: {
+				mockResolve: async (hook) => {
+					const noteID = hook.input.id;
+					const text = notes[noteID];
+					if (text != null) {
+						return {
+							data: {
+								notes_getNotesNoteByNoteID: {
```
Comment (jensneuse) [inline examples/openapi-mocking/.wundergraph/wundergraph.server.ts:18]: needs update as well after changing operationId

File: examples/openapi-mocking/.wundergraph/wundergraph.server.ts [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+							notes_postNotesNew: {
+								id: nextID,
+							},
+						},
+					};
+				},
+			},
+		},
+	},
+	graphqlServers: [
```
Comment (jensneuse) [inline examples/openapi-mocking/.wundergraph/wundergraph.server.ts:52]: can be removed.

File: examples/openapi-mocking/README.md [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+#### Create a new note
+
+```shell
+curl -d text=Hello http://localhost:9991/operations/NewNote
+```
+
+Notice the returned JSON with the newly created ID:
+
+```json
+{ "data": { "notes_postNotesNew": { "id": 1 } } }
```
Comment (jensneuse) [inline examples/openapi-mocking/README.md:24]: update and add info how to set operationId

**Review by jensneuse — CHANGES_REQUESTED** [review]

File: examples/openapi-mocking/.wundergraph/wundergraph.server.ts (hunk 4 above)
Comment (fiam) [reply examples/openapi-mocking/.wundergraph/wundergraph.server.ts]: Good catch, thanks!

File: examples/openapi-mocking/.wundergraph/api.yaml (hunk 1 above)
Comment (fiam) [reply examples/openapi-mocking/.wundergraph/api.yaml:16]: I've added it in the 2 places you suggested and updated the code accordingly. PTAL

**Review by StarpTech — APPROVED** [review]
LGTM

**Pull request merged**